tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,138 @@
+"""
+karapace - schema registry authentication and authorization tests
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.client import Client
+from karapace.schema_models import SchemaType, ValidatedTypedSchema
+from tests.utils import new_random_name, schema_avro_json, schema_jsonschema_json
+from typing import List
+from urllib.parse import quote
+
+import aiohttp
+import asyncio
+import requests
+
+admin = aiohttp.BasicAuth("admin", "admin")
+aladdin = aiohttp.BasicAuth("aladdin", "opensesame")
+reader = aiohttp.BasicAuth("reader", "secret")
+
+
+async def test_sr_auth(registry_async_client_auth: Client) -> None:
+    subject = new_random_name("cave-")
+
+    res = await registry_async_client_auth.post(f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json})
```

Comment:
`quote(subject)` is not needed, `subject` works fine. The format is something like `cave-5cfff223` that doesn't need `quote` call:

```
quote("cave/5cfff223")
'cave/5cfff223'
```

Calling `quote` makes things look like it's needed.

Reply:
Using `quote` is correct here, although `new_random_name` does not currently generate special characters. However if there would be any such special characters in subject name, then `quote` does the right thing.